crates/core/tedge_agent/src/agent.rs

```diff
@@ -225,10 +225,24 @@ impl SmAgent {
         })
     }
 
+    pub fn log_dir(&self) -> PathBuf {
+        self.config.log_dir.clone()
+    }
+
     #[instrument(skip(self), name = "sm-agent")]
-    pub async fn init(&mut self, config_dir: PathBuf) -> Result<(), anyhow::Error> {
+    pub async fn init(
+        &mut self,
+        config_dir: PathBuf,
+        log_path: PathBuf,
```

Comment:
Is this new `log_path` argument necessary? `self.log_dir()` should make it. Isn't it?